server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocatorTests.java

```diff
@@ -333,19 +304,23 @@ private RoutingAllocation createRoutingAllocation(ClusterState clusterState) {
         );
     }
 
-    private static ClusterState stateWithStartedIndices(List<IndexMetadata.Builder> indices) {
+    private static ClusterState stateWithStartedIndices(IndexMetadata.Builder... indices) {
         var metadataBuilder = Metadata.builder();
         var routingTableBuilder = RoutingTable.builder();
         for (var index : indices) {
             var inSyncId = UUIDs.randomBase64UUID(random());
-            var build = index.putInSyncAllocationIds(0, Set.of(inSyncId)).build();
+            var build = index.settings(settings(Version.CURRENT).put("index.routing.allocation.exclude._id", "node-3"))
```

Comment:
is this setting used intentionally here?

Reply:
Yes, It forces the shards to be rebalanced by a followup allocate call, I would add a comment